ahk/_hotkey.py

```diff
@@ -66,6 +73,9 @@ def __init__(self, executable_path: str, default_ex_handler: Optional[Callable[[
         self._get_callback_registry = functools.lru_cache(maxsize=None)(self._callback_registry_uncached)
         self._clipboard_callback: Optional[Callable[[int], Any]] = None
         self._clipboard_ex_handler: Optional[Callable[[int, Exception], Any]] = None
+        if directives is None:
+            directives = []
+        self._directives: list[Directive | Type[Directive]] = directives
```

Comment:
Maybe consider pre-computing which directives apply to the hotkey process so this has to be done one time and the `{% if ... %}` condition can be removed from the jinja template.

```suggestion
        self._directives: list[Directive | Type[Directive]] = [d for d in directives if d.apply_to_hotkeys_process]
```